src/main/capi/logical_types-c.cpp

```diff
@@ -204,6 +204,24 @@ char *duckdb_enum_dictionary_value(duckdb_logical_type type, idx_t index) {
 	return strdup(duckdb::StringValue::Get(value).c_str());
 }
 
+idx_t duckdb_enum_values(duckdb_logical_type type, char **values, idx_t length) {
```

Comment:
Can we add a check and a test for passing in `values` as NULL?

Perhaps the returned length should be an out-param instead and this function should return a `duckdb_state`?

Reply:
Ok, I changed this. Added a null check and test for it.